File: solutions/sonnet-1/sonnet/ruby/wordfreq.rb
text = STDIN.read
counts = Hash.new(0)
text.scan(/[A-Za-z]+/).each do |word|
  counts[word.downcase] += 1
end

counts.sort_by { |word, count| [-count, word] }.each do |word, count|
  puts "#{word} #{count}"
end

File: solutions/sonnet-1/sonnet/python/wordfreq.py
import sys
import re
from collections import Counter


def main():
    text = sys.stdin.read()
    words = re.findall(r"[A-Za-z]+", text)
    counts = Counter(w.lower() for w in words)
    for word, count in sorted(counts.items(), key=lambda kv: (-kv[1], kv[0])):
        print(f"{word} {count}")


if __name__ == "__main__":
    main()
